x-template.js

```diff
@@ -875,11 +868,22 @@ class TemplateEngine {
   }
 
   static #createRawResult(language, strings, values) {
-    return { [TemplateEngine.#LANGUAGE]: language, strings, values };
+    const analysis = TemplateEngine.#setIfMissing(TemplateEngine.#stringsToAnalysis, strings, () => ({}));
```

Comment:
This is the part that is a potential bottleneck. I’m not seeing any immediate slowdowns in performance — BUT, the performance test really isn’t testing the extremes of this. It’s essentially testing the performance of getting a value from a map with a size of _one_ (since we’re updating with the _same_ strings — that’s what an update is…). It’s possible that when we have a very _large_ number of templates, this could become less performant. We will have _at least_ as many items in this map as there are components in the page (assuming that a component invokes `html` only once — which is typically an under-estimate).